bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs

```diff
@@ -136,4 +135,40 @@ public async Task SoftDeleteManyByIdAsync(IEnumerable<Guid> ids)
             await dbContext.SaveChangesAsync();
         }
     }
+
+    public async Task<IEnumerable<Core.SecretsManager.Entities.Secret>> ImportAsync(IEnumerable<Core.SecretsManager.Entities.Secret> secrets)
+    {
+        try
+        {
+            using (var scope = ServiceScopeFactory.CreateScope())
+            {
+                var dbContext = GetDatabaseContext(scope);
+                var entities = new List<Secret>();
+                var projects = secrets.SelectMany(s => s.Projects?.Select(p => Mapper.Map<Project>(p)) ?? new List<Project>()).DistinctBy(p => p.Id).ToList();
```

Comment:
We might want to use a dictionary here for performance.

Reply:
Do you mind expanding a bit how we could use a dictionary here to increase performance? I don't feel confident using the dictionary solutions I came up with due to a lack of readability. However, this might definitely be a spot we want to make sure the performance is more finely tuned.

Reply:
On line 157 we use `projects.Find`. That operation is `O(n)`. Since we're doing it in a loop, that would be `O(n*m)`. If we use a Dictionary for projects it's `O(m)`.

Reply:
```suggestion
                var projects = secrets
                    .SelectMany(s => s.Projects ?? Enumerable.Empty<Project>)
                    .DistinctBy(p => p.Id)
                    .Select(p => Mapper.Map<Project>(p))
                    .ToDictionary(p => p.Id, p => p);
```

Reply:
Updated, thanks for the explanation and recommendation!